lib/kafka/sasl_authenticator.rb

```diff
@@ -31,7 +33,12 @@ def initialize(logger:, sasl_gssapi_principal:, sasl_gssapi_keytab:,
         logger: @logger,
       )
 
-      @mechanism = [@gssapi, @plain, @scram].find(&:configured?)
+      @oauth = Sasl::OAuth.new(
+        token_provider: sasl_oauth_token_provider,
+        logger: @logger,
```

Comment:
trailing coma not needed

Reply:
I tried to keep the same style as the other constructors with the trailing commas! In general I believe these are just to minimize git commit lines if another parameter is added here.